### PR TITLE
Refactoring repeating strings and text in buttons

### LINKED ISSUE #54: Orginize repeating strings like text in alerts and button text.
Create enums for repeating strings like text in alerts and button text for both flows such as:
```swift
    private let alertLogOutTitle = "Do you really want to log out?"
    private let navigationTitle = "My profile"

    private let cancelButtonLabelText = "Cancel"
    private let continueButtonLabelText = "Continue"
    private let saveButtonLabelText = "Save"
    private let editButtonLabelText = "Edit"
    private let logoutButtonLabelText = "Logout"
```

### PR DESCRIPTION
**Pull request by iBYarchitect**
## Summary
This PR is a part of the work to resolve #54 

## Changes
- Added enum AppAlert
- Added enum AppButton

## Checks to complete
- [x] The PR title has a conventional commit style (short and descriptive).
- [x] The PR description includes essential information that helps understand the purpose of the changes.
- [x] I've built and run my changes, and no warnings or errors occurred.
- [x] All existing tests passed with my changes.
- [x] My code follows our style guide.
- [x] I've performed a self-review of my code.
- [x] Every non-private interface is documented properly.
- [x] I've added tests for my code (if it makes sense).
- [x] I've updated `CHANGELOG.md` file.
- [x] PR has assignee, reviewers, milestone, it is properly linked to the project and to the issue.

### CODE REVIEW

File: Woof/Woof/Helper/AppAlert/AppAlert.swift [hunk 1]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
```
Comment (kirpichonok) [inline Woof/Woof/Helper/AppAlert/AppAlert.swift:2]: It's superfluous. See the rest of code.
```suggestion
```

File: Woof/Woof/Helper/AppAlert/AppAlert.swift [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+/// Represents alerts which uses in application.
```
Comment (kirpichonok) [inline Woof/Woof/Helper/AppAlert/AppAlert.swift:3]: nit:
```suggestion
/// Represents alert that is used in the application.
```

File: Woof/Woof/Helper/AppAlert/AppAlert.swift [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    /// Alert shown as confirmation of log out
+    case logOut = "Do you really want to log out?"
+
+    /// Alert shown as a result of an error
+    case error = "Error"
+
+    /// Text description of alert
+    var description: String {
+        rawValue
+    }
```
Comment (kirpichonok) [inline Woof/Woof/Helper/AppAlert/AppAlert.swift:14]: As an allowed variant but up to you:
```suggestion
    /// Alert shown as confirmation of log out
    case logOut = "Do you really want to log out?"
    /// Alert shown as a result of an error
    case error = "Error"
    /// Text description of alert
    var description: String {
        rawValue
    }
```

File: Woof/Woof/Helper/AppAlert/AppAlert.swift [hunk 4]
```diff
@@ -0,0 +1,15 @@
+import Foundation
+
+/// Represents alerts which uses in application.
+enum AppAlert: String, CustomStringConvertible, CaseIterable {
```
Comment (kirpichonok) [inline Woof/Woof/Helper/AppAlert/AppAlert.swift:4]: Why do we need an _enum_? It's important to be sure that we choose among restrict amount of cases? I don't really understand a purpose to adopt `CustomStringConvertible`. It seems as an overengineering.

File: Woof/Woof/Helper/AppButton/AppButton.swift [hunk 5]
```diff
@@ -0,0 +1,27 @@
+import Foundation
+
+/// Represents name of buttons which uses in application.
```
Comment (kirpichonok) [inline Woof/Woof/Helper/AppButton/AppButton.swift:3]: nit:
```suggestion
/// Represents name of button which is used in the application.
```

File: Woof/Woof/Helper/AppButton/AppButton.swift [hunk 6]
```diff
@@ -0,0 +1,27 @@
+import Foundation
+
```
Comment (kirpichonok) [inline Woof/Woof/Helper/AppButton/AppButton.swift:2]:
```suggestion
```

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift [hunk 7]
```diff
@@ -20,7 +20,7 @@ struct OwnerProfileView: View {
                             address: $viewModel.address
                         )
 
-                        Button(saveButtonLabelText) {
+                        Button(AppButton.save.description) {
```
Comment (kirpichonok) [inline Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift:23]: See the rest of code.
Screenshot 2023-12-14 at 14 57 53

File: Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift
Comment (kirpichonok) [inline Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift:1]: These tests seem to be redundant. You don't actually need this behavior from this type.

**Review by kirpichonok — CHANGES_REQUESTED** [review]
It's difficult to start again, but it's necessary! 🙂
Good attempt, but keep it simple.

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 3 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: Got it, but I'm save the current version.

File: Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift
Comment (iBYarchitect) [reply Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift:1]: In my opinion, these tests check possible sudden changes in descriptions.

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 2 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: Fixed with f4f7a7c

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 1 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: Fixed with f4f7a7c

File: Woof/Woof/Helper/AppButton/AppButton.swift (hunk 5 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppButton/AppButton.swift]: Fixed with f4f7a7c

File: Woof/Woof/Helper/AppButton/AppButton.swift (hunk 6 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppButton/AppButton.swift]: Fixed with f4f7a7c

File: Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift (hunk 7 above)
Comment (iBYarchitect) [reply Woof/Woof/View/OwnerFlow/OwnerProfileView/OwnerProfileView.swift]: Fixed with f4f7a7c

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 4 above)
Comment (NadzeyaShpakouskaya) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: Agree with @kirpichonok that we can use struct. We don't have to call the rawValue to get access to the string or make it static let in enum as an option. WDY?

File: Woof/Woof/Helper/AppAlert/AppAlert.swift [hunk 8]
```diff
@@ -0,0 +1,8 @@
+/// Represents the alerts that are used in the application.
+enum AppAlert: String, CaseIterable {
```
Comment (NadzeyaShpakouskaya) [inline Woof/Woof/Helper/AppAlert/AppAlert.swift:2]: Alerts have a title and a message. Will this enum cover both cases? And as discussed earlier do we really need the enum with the restricted cases?

File: Woof/WoofTests/Helper/AppButton/AppButtonDescriptionTests.swift [hunk 9]
```diff
@@ -0,0 +1,21 @@
...
+        // Given
+        let cancel = "Cancel"
+        let save = "Save"
+        let edit = "Edit"
+        let tryAgain = "Try Again"
+        let logout = "Logout"
+        let `continue` = "Continue"
+
+        // When // Then
+        XCTAssertEqual(AppButton.cancel.rawValue, cancel)
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/Helper/AppButton/AppButtonDescriptionTests.swift:14]: Could we simplify it? We do not have to create  constants and call them only once as we can compare values in asserts. WDY?
```suggestion
        // Given
        let cancel = "Cancel"
        let save = "Save"
        let edit = "Edit"
        let tryAgain = "Try Again"
        let logout = "Logout"
        let `continue` = "Continue"

        // When // Then
        XCTAssertEqual(AppButton.cancel.rawValue, "Cancel")
```

File: Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift [hunk 10]
```diff
@@ -0,0 +1,13 @@
+import XCTest
+
+final class AppAlertDescriptionTests: XCTestCase {
+    func testAppAlertDescriptionReturnsExpectedString() {
+        // Given
+        let logOut = "Do you really want to log out?"
+        let error = "Error"
+
+        // When // Then
+        XCTAssertEqual(AppAlert.logOut.rawValue, logOut)
```
Comment (NadzeyaShpakouskaya) [inline Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift:10]: Could we simplify it? We do not have to create constants and call them only once as we can compare values in asserts. WDY?
```suggestion
        let logOut = "Do you really want to log out?"
        let error = "Error"

        // When // Then
        XCTAssertEqual(AppAlert.logOut.rawValue, "Do you really want to log out?")
```

**Review by NadzeyaShpakouskaya — CHANGES_REQUESTED** [review]
Consider minor improvements

File: Woof/Woof/Helper/AppButton/AppButton.swift (hunk 5 above)
Comment (kirpichonok) [reply Woof/Woof/Helper/AppButton/AppButton.swift]: As far as I know, it's common agreement to use singular nouns in the documentation.

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 4 above)
Comment (kirpichonok) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: We can use _static_ properties of an _Enum_ or a _Structure_. These constants are not strictly restricted set.

**Review by kirpichonok — COMMENTED** [review]
I agree with @NadzeyaShpakouskaya. Try one more time.

File: Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift (hunk 10 above)
Comment (iBYarchitect) [reply Woof/WoofTests/Helper/AppAlert/AppAlertDescriptionTests.swift]: Fixed with 03725e9

File: Woof/WoofTests/Helper/AppButton/AppButtonDescriptionTests.swift (hunk 9 above)
Comment (iBYarchitect) [reply Woof/WoofTests/Helper/AppButton/AppButtonDescriptionTests.swift]: Fixed with 03725e9

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 8 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: Agreed I replaced cases with static properties.

File: Woof/Woof/Helper/AppButton/AppButton.swift [hunk 11]
```diff
@@ -0,0 +1,20 @@
+/// Represents the names of buttons used in the application.
+enum AppButton: String, CaseIterable {
+    /// Represents the "Cancel" button.
+    case cancel = "Cancel"
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppButton/AppButton.swift]: Fixed with 03725e9

File: Woof/Woof/Helper/AppButton/AppButton.swift [hunk 12]
```diff
@@ -0,0 +1,20 @@
+/// Represents the names of buttons used in the application.
+enum AppButton: String, CaseIterable {
```
[the remark this answers is not in the document]
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppButton/AppButton.swift]: Yep, we mean button titles.

File: Woof/Woof/Helper/AppButton/AppButton.swift (hunk 5 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppButton/AppButton.swift]: Fixed with 03725e9

File: Woof/Woof/Helper/AppAlert/AppAlert.swift (hunk 4 above)
Comment (iBYarchitect) [reply Woof/Woof/Helper/AppAlert/AppAlert.swift]: Fixed with 03725e9

**Review by NadzeyaShpakouskaya — COMMENTED** [review]
LGTM

**Review by NadzeyaShpakouskaya — APPROVED** [review]

**Pull request merged**